R/03-sbc-centric-stats.R

```diff
@@ -0,0 +1,535 @@
+  var <- "drive_dist_mean"
+  loc_col <- "nearest_facility"
+  plot_title <- glue("Driving Distances for {loc_id}")
+  var_title <- "Driving Distance (km)"
+  
+  map_plot <- build_map(
+    data = map_data,
+    servicebc_data = sbc_locs,
+    varname = var,
+    csd_name = loc_id,
+    csd_col = loc_col,
+    map_theme = MAP_THEME,
+    fill_scale = FILL_THEME,
+    plot_title = plot_title,
+    legend_title = var_title
+  )
+  
+  # Save the map
+  ggsave(
+    filename = glue("{map_folder}/{loc_id}_drive_distance_map.png"),
+    plot = map_plot,
+    width = 10,
+    height = 8,
+    dpi = 300
+  )
+}
+
+# Display the map for a specific location as an example
+loc_id <- unique(map_data$nearest_facility)[1]
+var <- "drive_dist_mean"
+loc_col <- "nearest_facility"
+plot_title <- glue("Driving Distances for {loc_id}")
+var_title <- "Driving Distance (km)"
+
+example_map <- build_map(
+  data = map_data,
+  servicebc_data = sbc_locs,
+  varname = var,
+  csd_name = loc_id,
+  csd_col = loc_col,
+  map_theme = MAP_THEME,
+  fill_scale = FILL_THEME,
+  plot_title = plot_title,
+  legend_title = var_title
+)
+
+print(example_map)
+
+# =========================================================================== #
+# Population Pyramid Creation ----
+# =========================================================================== #
+
+# Prepare data for population pyramids by joining with location assignment data
+pyramid_data <- drivetime_data_reduced |>
+  distinct(assigned, dbid) |>
+  inner_join(
+    db_projections_transformed |>
+      filter(year %in% c(CURRENT_YEAR, CURRENT_YEAR + 5, CURRENT_YEAR + 10)),
+    by = "dbid"
+  )
+
+# Get unique facility names to generate pyramids for
+facilities <- unique(pyramid_data$assigned)
+
+# Generate population pyramids for each facility
+population_pyramids <- list()
+for (facility in facilities) {
+  population_pyramids[[facility]] <- create_population_pyramid(
+    data = pyramid_data,
+    location_name = facility
+  )
+}
+
+# Display a sample pyramid for the first facility
+if (length(facilities) > 0) {
+  print(population_pyramids[[facilities[2]]])
+}
+
+# Create a folder for drive distance maps
+pyramid_folder <- file.path(MAP_OUT, "sbc_population_pyramids")
+if (!dir.exists(pyramid_folder)) {
+  dir.create(pyramid_folder, recursive = TRUE)
+}
+
+for (facility in facilities) {
+  ggsave(
+    filename = glue("{pyramid_folder}/{facility}_population_pyramid.png"),
+    plot = population_pyramids[[facility]],
+    width = 10,
+    height = 8,
+    dpi = 300
+  )
+}
+
+# Create a combined view with multiple pyramids
+# Take up to 4 facilities for a nice grid layout
+if (length(facilities) > 1) {
+  combined_pyramids <- cowplot::plot_grid(
+    plotlist = population_pyramids[1:min(4, length(facilities))],
+    ncol = 2
+  )
+  
+  ggsave(
+    filename = glue("{pyramid_folder}/combined_population_pyramids.png"),
+    plot = combined_pyramids,
+    width = 16,
+    height = 12,
+    dpi = 300
+  )
+}
```

Comment:
There are a few of these calculations you could make more robust.
```suggestion
    est_pop = sum(population, na.rm = TRUE),
```

Reply:
fixed in new file